magma/bits.py

```diff
@@ -456,6 +456,13 @@ def undriven(self):
             raise TypeError("undriven can only be used on input")
         m.wire(DefineUndriven(len(self))().O, self)
 
+    def as_bits(self):
```

Comment:
should all these functions (`as_bits`, `from_bits`, etc.) be added as methods to the metaclasses?

Reply:
`as_bits` needs to be a a function on a value (given a value, return me the bits representation).  We could add an interface to convert a meta class (type) to a Bits, but we would maybe want to use a different name to clearly distinguish the behavior. (e.g. as_Bits or to_Bits).

`from_bits` is a classmethod on the type, so I can say something like `T.from_bits`, so create me a T from an instance of Bits.  I think this is a reasonable organization.